### PR TITLE
Add linting

### PR DESCRIPTION
**Pull request by arjunnaha**

### CODE REVIEW

File: frontend/src/ConfigManager.js [hunk 1]
```diff
@@ -6,11 +7,12 @@ const DEFAULT_ID = 0;
 class ConfigManager {
   constructor() {
     this.graphID = DEFAULT_ID;
+    // eslint-disable-next-line global-require
     this.setConfig(require("./config.json"));
   }
 
   getGraphID = () => {
-    return this.graphID++;
+    return this.graphID + 1;
```
Comment (MiltFra) [inline frontend/src/ConfigManager.js:15]: This doesn't look equivalent to me. Are you sure we don't want to update `this.graphID` as well?

File: frontend/src/ConfigManager.js [hunk 2]
```diff
@@ -95,20 +94,12 @@ class ConfigManager {
...
-          problems.push(
-            `Invalid field in graph ${graphNum}: "paths" should be an array of arrays`
-          );
-        } else if (
-          !graph.paths.every(path => path.every(key => typeof key === "string"))
-        ) {
-          problems.push(
-            `Invalid field in graph ${graphNum}: "paths" should be an array of string arrays`
-          );
+        if (!graph.paths.every((path) => Array.isArray(path))) {
```
Comment (MiltFra) [inline frontend/src/ConfigManager.js:99]: I don't know wth is going on here...

File: frontend/src/components/ButtonContainer/ButtonContainerDebug.js [hunk 3]
```diff
@@ -103,22 +99,18 @@ export default function ButtonContainerDebug(props) {
...
-  return (
-    <div className="button-container">
-      {getButtons(props.debugData.isCompiled)}
-    </div>
-  );
-};
+    return [getButton(button)];
+  };
+
+  // eslint-disable-next-line react/destructuring-assignment
```
Comment (MiltFra) [inline frontend/src/components/ButtonContainer/ButtonContainerDebug.js:114]: I think the old one looked nicer tbh

File: frontend/src/components/DataContainer/DataContainer.js [hunk 4]
```diff
@@ -1,57 +1,54 @@
...
+
+import DataList from "../DataList/DataList";
+import DataRow from "../DataRow/DataRow";
+
+export default function DataContainer({ telemetryData }) {
+  const dataList = {};
 
   const combine = (title, data) => {
-    for (var list of Object.values(data)) {
+    // eslint-disable-next-line no-restricted-syntax
```
Comment (MiltFra) [inline frontend/src/components/DataContainer/DataContainer.js:15]: Why are we disabling these rules? Either follow them or remove them.

File: frontend/src/components/DataList/DataList.js [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faCaretUp, faCaretDown } from "@fortawesome/free-solid-svg-icons";
 import { Collapse } from "react-collapse";
 
-export default function DataList(props) {
+export default function DataList({ title, value }) {
   const [isOpen, setIsOpen] = useState(true);
   return (
     <div>
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
```
Comment (MiltFra) [inline frontend/src/components/DataList/DataList.js:13]: You're mixing comment styles as well. Stick with one.

File: frontend/src/components/DataPointSelector/DataPointSelector.js [hunk 6]
```diff
@@ -1,48 +1,46 @@
...
   const [paths, setPaths] = useState();
 
   useEffect(() => {
-    setPaths(getAllPaths(props.telemetryData));
-  }, [props.telemetryData]);
+    setPaths(getAllPaths(telemetryData));
+  }, [telemetryData]);
 
   const getPathList = () => {
     // const paths = ;
```
Comment (MiltFra) [inline frontend/src/components/DataPointSelector/DataPointSelector.js:20]: ?

File: frontend/src/components/GraphsContainer/GraphsContainer.js [hunk 7]
```diff
@@ -22,114 +25,104 @@ export default function GraphsContainer(props) {
...
-      setCurrentGraph(-1);
-    };
-  
-    const handleDataPointClicked = path => {
-      ConfigManager.handlePath(path, currentGraph);
-    };
-  
-    const isSelected = path => {
-      return ConfigManager.isPathSelected(path, currentGraph);
+  // const handleToolClick =() => {
```
Comment (MiltFra) [inline frontend/src/components/GraphsContainer/GraphsContainer.js:60]: Dead code...

File: frontend/src/react-app-env.d.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (MiltFra) [inline frontend/src/react-app-env.d.ts:1]: what's this?

File: frontend/src/serviceWorker.js [hunk 9]
```diff
@@ -11,17 +11,15 @@
...
-  window.location.hostname === 'localhost' ||
+  window.location.hostname === "localhost" ||
     // [::1] is the IPv6 localhost address.
-    window.location.hostname === '[::1]' ||
+    window.location.hostname === "[::1]" ||
     // 127.0.0.0/8 are considered localhost for IPv4.
-    window.location.hostname.match(
-      /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/
-    )
+    window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/),
```
Comment (MiltFra) [inline frontend/src/serviceWorker.js:18]: What's up with this regex?

File: frontend/src/serviceWorker.js [hunk 10]
```diff
@@ -11,17 +11,15 @@
...
 );
 
 export function register(config) {
-  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+  if (process.env.NODE_ENV === "production" && "serviceWorker" in navigator) {
```
Comment (MiltFra) [inline frontend/src/serviceWorker.js:22]: the precedence here scares me...

**Review by MiltFra — CHANGES_REQUESTED** [review]

File: frontend/src/ConfigManager.js (hunk 1 above)
Comment (arjunnaha) [reply frontend/src/ConfigManager.js:15]: Ah yes you're very right.

File: frontend/src/ConfigManager.js (hunk 2 above)
Comment (MiltFra) [reply frontend/src/ConfigManager.js:99]: Actually, upon re-reading this, it actually makes sense.

**Comment by arjunnaha** [comment]
As we are moving towards OpenMCT for next year and the lack of any tests to verify the functionality is preserved, this PR is going to be closed without merging :(

**Pull request closed without merging**